### PR TITLE
Hyperlink fixes

### PR DESCRIPTION
**Pull request by Bcorde5**
Fixed issues brought up in #1817 , please report any additional problems, and don't be afraid to ask questions, I am sure there will be some.

### CODE REVIEW

File: tools/inspect/apple_macro_check.cpp [hunk 1]
```diff
@@ -90,8 +92,6 @@ namespace boost
                   line_start = it + 1; // could be end()
               }
           }
-
-          std::string lineloc = linelink(full_path, std::to_string(line_number));
```
Comment (hkaiser) [inline tools/inspect/apple_macro_check.cpp]: Why was this removed? Shouldn't the line number be represented by a hyperlink?

**Comment by hkaiser** [comment]
The `ascii_check` checker still does not create the proper hyperlink for the given line number.

File: tools/inspect/apple_macro_check.cpp (hunk 1 above)
Comment (Bcorde5) [inline tools/inspect/apple_macro_check.cpp]: The line number should be represented by a hyperlink, however the structure of the inspect tool requires the line_number to be a size_t, so I implemented the switch in inspect.cpp for this particular case, when there are four arguments in the error report.

Comment (hkaiser) [inline tools/inspect/apple_macro_check.cpp]: Why don't you use this overload for all of the checkers, then? What's so special about `apple_macro_check`?

Also, where is that `switch` you're referring to?

File: tools/inspect/inspect.cpp [hunk 2]
```diff
@@ -1062,7 +1077,8 @@ void print_output(std::ostream& out, inspector_list const& inspectors)
       "<tr>\n"
       "<td>"
       "<a href = \"https://github.com/STEllAR-GROUP/hpx\">"
-      "<img src=\"http://stellar.cct.lsu.edu/files/stellar100.png\" alt=\"STE||AR logo\" />"
+      "<img src=\"http://stellar.cct.lsu.edu/files/stellar100.png\"\
+        alt=\"STE||AR logo\" />"
```
Comment (hkaiser) [inline tools/inspect/inspect.cpp]: Please don't break a quoted string in the middle by inserting a `\<newline>`. This is just confusing, Use compile time string concatination instead:

    ""

**Comment by hkaiser** [comment]
Because of the GIT hash not being available at compile time when compiling in CircleCI, we should also add a command line option to the inspect tool which allows to specify the hash to be used on the tool's command line. Something like:

    inspect --git-hash=eda56f5255eb9a27a7e3d1f2daffe5a1ac3bbc1f ...

Internally, the inspect tool should use the HPX_HAVE_GIT_COMMIT constant by default, but prefer using the command line argument if given.

File: tools/inspect/apple_macro_check.cpp (hunk 1 above)
Comment (Bcorde5) [inline tools/inspect/apple_macro_check.cpp]: It works for both apple_macro and ascii check, as both have four arguments, the last one having three.  If the link works, https://github.com/Bcorde5/hpx/blob/HyperlinkFixes/tools/inspect/inspect.cpp#L547-L555 is where the overload occurs.  This only works, and the only reason it must work is because in both of these checkers, the line number is not included in the string, but rather as a size_t argument.  The other checkers cannot use a similar system because they either do not have a line number or the line number is within the string, which makes it easier to point out multiple mistakes in a single run on inspect, but makes it tricky creating such an overload for the other checkers with only three arguments.
